aicsimageio/aics_image.py

```diff
@@ -0,0 +1,418 @@
+        """
+        Get specific dimension image data out of an image as a numpy array.
+
+        Parameters
+        ----------
+        out_orientation: Optional[str]
+            A string containing the dimension ordering desired for the returned ndarray.
+            Default: The current image dimensions. i.e. `self.dims.order`
+
+        kwargs: Any
+            * C=1: specifies Channel 1
+            * T=3: specifies the fourth index in T
+            * D=n: D is Dimension letter and n is the index desired. D should not be
+              present in the out_orientation.
+            * D=[a, b, c]: D is Dimension letter and a, b, c is the list of indicies
+              desired. D should be present in the out_orientation.
+            * D=(a, b, c): D is Dimension letter and a, b, c is the tuple of indicies
+              desired. D should be present in the out_orientation.
+            * D=range(...): D is Dimension letter and range is the standard Python
+              range function. D should be present in the out_orientation.
+            * D=slice(...): D is Dimension letter and slice is the standard Python
+              slice function. D should be present in the out_orientation.
+
+        Returns
+        -------
+        data: numpy array
+            The image data with the specified dimension ordering.
+
+        Examples
+        --------
+        Specific index selection
+
+        >>> img = AICSImage("s_1_t_1_c_10_z_20.ome.tiff")
+        ... c1 = img.get_image_data("ZYX", C=1)
+
+        List of index selection
+
+        >>> img = AICSImage("s_1_t_1_c_10_z_20.ome.tiff")
+        ... first_and_second = img.get_image_data("CZYX", C=[0, 1])
+
+        Tuple of index selection
+
+        >>> img = AICSImage("s_1_t_1_c_10_z_20.ome.tiff")
+        ... first_and_last = img.get_image_data("CZYX", C=(0, -1))
+
+        Range of index selection
+
+        >>> img = AICSImage("s_1_t_1_c_10_z_20.ome.tiff")
+        ... first_three = img.get_image_dask_data("CZYX", C=range(3))
+
+        Slice selection
+
+        >>> img = AICSImage("s_1_t_1_c_10_z_20.ome.tiff")
+        ... every_other = img.get_image_data("CZYX", C=slice(0, -1, 2))
+
+        Notes
+        -----
+        * If a requested dimension is not present in the data the dimension is
+          added with a depth of 1.
+
+        See `aicsimageio.transforms.reshape_data` for more details.
+        """
+        pass
+
+    @property
+    def metadata(self) -> Any:
+        """
+        Passthrough to the base image reader metadata property.
+        For more information, see the specific image format reader you are using
+        for details on it's metadata property.
+        """
+        pass
+
+    def get_channel_names(self, scene: int = 0) -> List[str]:
+        """
+        Attempts to use the available metadata for the image to return the channel
+        names for the image.
+
+        If no channel metadata is available, returns a list of string indicies for each
```

Comment:
need to replace `indicies` with `indices` throughout.  it shows up in a few places